packages/client/wallets/smart-wallet/src/blockchain/wallets/passkey.ts

```diff
@@ -37,57 +36,62 @@ export class PasskeyAccountService {
+
+        const passkey = await this.getPasskey({ user, walletParams }, existingSignerConfig);
+        const validator = await toPasskeyValidator(publicClient, {
+            webAuthnKey: passkey,
+            entryPoint: entryPoint.address,
+            validatorContractVersion,
+            kernelVersion,
+        });
 
         const kernelAccount = await createKernelAccount(publicClient, {
             plugins: { sudo: validator },
             entryPoint: entryPoint.address,
             kernelVersion,
         });
 
-        return { signerData: this.getSignerData(validator, walletParams.signer.passkeyName), account: kernelAccount };
+        return {
+            signerData: this.getSignerData(validator, validatorContractVersion, walletParams.signer.passkeyName),
+            account: kernelAccount,
+        };
     }
 
-    private async getValidator(
-        { user, entryPoint, publicClient, walletParams, kernelVersion }: Omit<PasskeyWalletParams, "chain">,
+    private async getPasskey(
+        { user, walletParams }: { user: UserParams; walletParams: WalletParams & { signer: PasskeySigner } },
         existingSignerConfig?: PasskeySignerData
-    ): Promise<PasskeyValidator> {
+    ): Promise<WebAuthnKey> {
         if (existingSignerConfig != null) {
-            return await deserializePasskeyValidator(publicClient, {
-                serializedData: serializePasskeyValidatorData(existingSignerConfig),
-                entryPoint: entryPoint.address,
-                kernelVersion,
-            });
+            return {
+                pubX: BigInt(existingSignerConfig.pubKeyX),
+                pubY: BigInt(existingSignerConfig.pubKeyY),
+                authenticatorId: existingSignerConfig.authenticatorId,
+                authenticatorIdHash: existingSignerConfig.authenticatorIdHash,
+            };
         }
 
-        const webAuthnKey = await toWebAuthnKey({
+        return toWebAuthnKey({
             passkeyName: walletParams.signer.passkeyName ?? "",
             passkeyServerUrl: this.crossmintService.getPasskeyServerUrl(),
             mode: WebAuthnMode.Register,
             passkeyServerHeaders: this.createPasskeysServerHeaders(user),
         });
-
-        return toPasskeyValidator(publicClient, {
-            webAuthnKey,
-            entryPoint: entryPoint.address,
-            kernelVersion,
-        });
     }
 
-    private getSignerData(validator: PasskeyValidator, passkeyName: string = ""): PasskeySignerData {
+    private getSignerData(
+        validator: PasskeyValidator,
+        validatorContractVersion: PasskeyValidatorContractVersion,
+        passkeyName: string = ""
+    ): PasskeySignerData {
         const fields = deserializePasskeyValidatorData(validator.getSerializedData());
+
         return {
             ...fields,
             passkeyName,
+            validatorContractVersion,
```

Comment:
The output of this function is stored in Crossmint's backend, we want to keep this new version around for when they introduce a new passkey validator and we need to keep compatibility with already deployed wallets, similar to entrypoint or kernel versions.